BHBC-981: Update create project page to wizard style

NickPhura opened the pull request:
# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-981

## This PR contains the following changes

- Move create project components to /create folder
- Add stepper wizard component
- Remove mouse scroll zoom on leaflet map

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

NickPhura committed:
BHBC-981: Update create project page to wizard style
- Move create project components to /create folder
- Add stepper wizard component
- Remove mouse scroll zoom on leaflet map

NickPhura commented:
I still need to write tests for the the StepperWizard component, but need to run to an appointment, so I'll have to do them later or tomorrow.  I can add them in this PR or a separate one, doesn't matter to me.

sdevalapurkar commented on app/src/features/projects/create/CreateProjectPage.tsx:
```diff
@@ -131,24 +129,34 @@ const CreateProjectPage: React.FC = () => {
   const [enableCancelCheck, setEnableCancelCheck] = useState(true);
 
-  // Tracks the active step #
+  // Tracks the active  #
```
nit:
I think step # makes sense

sdevalapurkar commented on app/src/features/projects/create/CreateProjectPage.tsx:
```diff
@@ -458,11 +523,11 @@ const CreateProjectPage: React.FC = () => {
    * Returns the step index for the first invalid form step, or `-1` if all steps are valid
    *
-   * @return {*} {number}
+   * @return {*} {number}F
```
```suggestion
   * @return {*} {number}
```

sdevalapurkar reviewed (commented):
looking nice :)

anissa-agahchen reviewed (commented):
Found a couple of issues, no sure that they are entirely caused by your PR ... but maybe caused by validation changes:

![Screen Shot 2021-05-03 at 4 45 46 PM](https://user-images.githubusercontent.com/22803003/116946658-90efb100-ac2f-11eb-8c30-9b76efae0970.png)

This one above is an issue if you try to edit the objectives section with an empty caveat.  If you touch the caveat, the issue goes away
![Screen Shot 2021-05-03 at 4 48 11 PM](https://user-images.githubusercontent.com/22803003/116946660-92b97480-ac2f-11eb-864d-ace3e31b4f14.png)

This 500 error occurred when I tried to save a funding source without an end-date

sdevalapurkar commented:
> I still need to write tests for the the StepperWizard component, but need to run to an appointment, so I'll have to do them later or tomorrow. I can add them in this PR or a separate one, doesn't matter to me.

I think separate PR would be better 😄

rstens committed:
Merge branch 'dev' into BHBC-981

NickPhura committed:
Code review updates:
- Updated some jsdoc comments
- Default caveats to empty string in API, when not populated (matches objectives)
- Add Required yup schema to funding source end_date
- Added StepperWizard tests

NickPhura committed:
Merge branch 'BHBC-981' of https://github.com/bcgov/biohubbc into BHBC-981

NickPhura commented:
> 
> 
> Found a couple of issues, no sure that they are entirely caused by your PR ... but maybe caused by validation changes:
> 

> 
> This one above is an issue if you try to edit the objectives section with an empty caveat. If you touch the caveat, the issue goes away

> 
> This 500 error occurred when I tried to save a funding source without an end-date

It looks like the issue was that the caveats field, in the API, was defaulting to `null` when no value was provided, rather than `empty string`. Updated to be empty string, which is what the objectives field was already doing.

sdevalapurkar commented on app/src/components/stepper-wizard/StepperWizard.test.tsx:
```diff
@@ -0,0 +1,323 @@
+import TextField from '@material-ui/core/TextField';
+import { cleanup, fireEvent, render, RenderResult, waitFor } from '@testing-library/react';
```
```suggestion
import { cleanup, fireEvent, render, waitFor } from '@testing-library/react';
```

sdevalapurkar commented on app/src/components/stepper-wizard/StepperWizard.test.tsx:
```diff
@@ -0,0 +1,323 @@
+const SampleFormikForm = () => {
+  const formikProps = useFormikContext<ISampleFormikFormProps>();
+
+  const { values, handleChange, handleSubmit } = formikProps;
```
```suggestion
  const { values, handleChange } = formikProps;
```

NickPhura committed:
Remove unused imports
update createProjectPage snapshot

sdevalapurkar reviewed (approved):
lgtm

anissa-agahchen commented:
Sorry, I found one more ... the validation is preventing me from creating a project without a permit
![Screen Shot 2021-05-04 at 3 52 39 PM](https://user-images.githubusercontent.com/22803003/117079610-f5267980-acf0-11eb-8ada-03a2e2e5f929.png)

sdevalapurkar commented:
> Sorry, I found one more ... the validation is preventing me from creating a project without a permit
> ![Screen Shot 2021-05-04 at 3 52 39 PM](https://user-images.githubusercontent.com/22803003/117079610-f5267980-acf0-11eb-8ada-03a2e2e5f929.png)

if you press add permit then it is supposed to
if you have no permits showing in the list then does it still prevent??

anissa-agahchen commented:
> > Sorry, I found one more ... the validation is preventing me from creating a project without a permit
> > ![Screen Shot 2021-05-04 at 3 52 39 PM](https://user-images.githubusercontent.com/22803003/117079610-f5267980-acf0-11eb-8ada-03a2e2e5f929.png)
> 
> if you press add permit then it is supposed to
> if you have no permits showing in the list then does it still prevent??

Actually, it's fine .. sorry for some reason I was expecting to not need those other two fields

anissa-agahchen reviewed (approved):
Looks really good
